python/ray/experimental/serve/queues.py

```diff
@@ -0,0 +1,155 @@
+from collections import defaultdict, deque
+from typing import Any
+
+import numpy as np
+from dataclasses import dataclass
+
+import ray
+from ray.experimental.serve.utils import get_custom_object_id, logger
+
+
+@dataclass
+class Query:
+    request_body: Any
+    result_oid: ray.ObjectID
+
+    @staticmethod
+    def new(req: Any):
+        return Query(request_body=req, result_oid=get_custom_object_id())
+
+
+@dataclass
+class WorkIntent:
+    work_oid: ray.ObjectID
+
+    @staticmethod
+    def new():
+        return WorkIntent(work_oid=get_custom_object_id())
+
+
+class CentralizedQueues:
+    """A "router" that routes request to available workers.
+
+    It aceepts requests from `.produce` method and queues the request data.
+    It also accepts work intention from workers via the `.consume` method.
+    The traffic policy is used to match requests with their corresponding
+    workers.
+
+    Behavior:
+        >>> # psuedo-code
+        >>> queue = CentralizedQueues()
+        >>> queue.produce('service-name', data)
+        # nothing happens, request is queued.
+        # returns result ObjectID, which will contains the final result
+        >>> queue.consume('backend-1')
+        # nothing happens, work intention is queued.
+        # return work ObjectID, which will contains the future request payload
+        >>> queue.link('service-name', 'backend-1')
+        # here the producer is matched with consumer, request data is put into
+        # work ObjectID, and the worker processs the request and store the
+        # result into result ObjectID
+
+    Traffic policy *splits* the traffic among different consumers. It behaves
+    in a probablistic fashion:
+
+    1. When all backends are ready to receive traffic, we will randomly choose
+       a backend based on the weights assigned by traffic policy dictionary.
+
+    2. When more than 1 but not all backends are ready, we will normalize the
+       weights of the ready backends to 1 and choose a backend via sampling.
+
+    3. When there is only 1 backend ready, we will directly use that backend.
```

Comment:
What does "directly" mean in this context? Sounds like it's using a different mechanism in this case. If not, just say "we will use only that backend."